connect/runtime/src/main/java/org/apache/kafka/connect/runtime/WorkerTask.java

```diff
@@ -132,7 +127,6 @@ public void stop() {
      */
     public void cancel() {
         cancelled = true;
-        retryWithToleranceOperator.triggerStop();
```

Comment:
I know it's a little strange for this class to hold onto the `RetryWithToleranceOperator` when its only interaction is to stop it, but this seemed reasonable in https://github.com/apache/kafka/pull/12478 because it saved us the potential headache of adding another subclass to `WorkerTask` that might forget to invoke `RetryWithToleranceOperator::triggerStop`.

I'm not hugely opposed to changing this part and it's not a blocker, but FWIW I think this change is unnecessary.

Reply:
I've added a commit that generifies WorkerTask, so that this can remain a field there in the superclass. I also moved up the error reporter supplier and transformation chain that were in a similar situation. Let me know what you think of those changes :)